src/internal/pachd/builder.go

```diff
@@ -548,6 +556,64 @@ func (b *builder) startDebugWorker(ctx context.Context) error {
 	return nil
 }
 
+func (b *builder) generatePJSWorkerAuthToken(ctx context.Context) error {
+	return generatePJSWorkerAuthToken(ctx, b.env.AuthServer(), b.env.GetKubeClient(), b.env.Config())
+}
+
+// TODO: move this function to the auth package(s) it can potentially be called again when we turn on auth.
+func generatePJSWorkerAuthToken(ctx context.Context, authServer auth.APIServer, kubeClient kubernetes.Interface,
+	config *pachconfig.Configuration) error {
+	secretName := "pachyderm-pjs-worker-authentication"
+	secretKey := "pjsWorkerAuthToken"
+	// First read the secret in case it exists.
+	existingSecret, err := kubeClient.CoreV1().Secrets(config.Namespace).Get(context.Background(), secretName, metav1.GetOptions{})
+	if err == nil {
+		// Secret exists, read the value from the specified key
+		if encodedValue, ok := existingSecret.Data[secretKey]; ok {
+			// Decode the value since it is stored as base64
+			token, err := base64.StdEncoding.DecodeString(string(encodedValue))
+			if err != nil {
+				return errors.Wrap(err, "decoding from base64")
+			}
+			if err := os.Setenv("PJS_WORKER_AUTH_TOKEN", string(token)); err != nil {
```

Comment:
Why set the environment variable here — isn’t the point of a secret that k8s sets in the container’s environment?

Reply:
You're right, I think we just need to set it in the config.

Kubernetes sets the container's environment, but in this case, the full pachd container doesn't have this configuration beforehand (because the full pachd is responsible for creating it if it doesn't exist)